Feat/ai label wc tanstack

devadula-nandan opened the pull request:
closes [#7353](https://github.com/carbon-design-system/ibm-products/issues/7353) on ibm-products

- adds ai label example.
- fixes ai-lable style on cells [#19293](https://github.com/carbon-design-system/carbon/pull/19293) in carbon-web-components. the `@carbon/web-components` package was on latest, so the fixes will be auto applied.
- fixes/refactors sorting example (to be compatible with ai-label) and also follows carbon wc data-table api/design
- adds selection example with ai-label

  - Note: ai label presence is tightly coupled with this prop `<cds-table-row selection-name="1">`. but tanstack has a different way of maintaining a select column in column definition. we might need to look for a better way to map these two behaviors together.

- adds sandboxes to readme

preview: [ai-label](https://stackblitz.com/github/devadula-nandan/tanstack-carbon/tree/feat/ai-label-wc-tanstack/web-components/ai-label), [sortable](https://stackblitz.com/github/devadula-nandan/tanstack-carbon/tree/feat/ai-label-wc-tanstack/web-components/sortable)
note: use `"@carbon/web-components": "^2.27.2",` as stackblitz might cache versions

matthewgallo commented on web-components/ai-label/src/with-selection.ts:
```diff
@@ -0,0 +1,282 @@
+`;
+
+const columns: ColumnDef<Resource, any>[] = [
+  // {
+  //   id: 'select',
+  //   header: ({ table }) => {
+  //     console.log(table.getIsAllRowsSelected());
+  //     return html`
+  //       <cds-checkbox
+  //         ?checked="${table.getIsAllRowsSelected()}"
+  //         .indeterminate="${table.getIsSomeRowsSelected()}"
+  //         @cds-checkbox-changed="${table.getToggleAllRowsSelectedHandler()}"></cds-checkbox>
+  //     `;
+  //   },
+  //   cell: ({ row }) => html`
+  //     <cds-checkbox
+  //       @cds-checkbox-changed='${row.getToggleSelectedHandler()}'
+  //       ?checked='${row.getIsSelected()}'
+  //       ?disabled='${!row.getCanSelect()}'
+  //       .indeterminate='${row.getIsSomeSelected()}'
+  //     /></cds-checkbox>
+  //   `,
+  // },
```
Can you remove if these columns aren't going to be included in the example?

devadula-nandan replied:
👍🏻 , we might need to update batch actions web-components example in a follow up. to use the data-table api, and sync the tanstack selection column from the instance, through the data-table api custom events. also in react `select all` checkbox selects the current page, and select all from toolbar selects the entire table. which i think is the right behavior, and the web components needs an update on this 🤔 

i am thinking of disabling the selection column from rendering, and let the data-table selection specific callbacks update the table state

<img width="701" alt="Screenshot 2025-05-12 at 1 33 06 PM" src="https://github.com/user-attachments/assets/1bf590cb-1179-40ee-b385-f0a92aacab4c" />

ill remove these comments for now, but will try to find a better selection handling with the expected behaviors in web-component, and will update this and batch actions example.

matthewgallo commented on web-components/ai-label/src/with-selection.ts:
```diff
@@ -0,0 +1,282 @@
+];
+
+const data: Resource[] = makeData(100, { aiLabelRows: [1, 3] });
+console.log(data);
```
Remove

matthewgallo commented on web-components/ai-label/src/with-selection.ts:
```diff
@@ -0,0 +1,282 @@
+      debugTable: true,
+    });
+
+    console.log(table.getRowModel().rowsById);
+
+    console.log(table.getState().rowSelection);
+    console.log(table.getRowCount());
+    console.log('are some rows selected: ', table.getIsSomeRowsSelected());
```
Remove console logs

devadula-nandan replied:
we might as well clean up the batch actions, as i took the base template from it.

devadula-nandan replied:
cleaned them from batch actions example as well. as i took it as a base template.

matthewgallo commented on web-components/ai-label/src/with-selection.ts:
```diff
@@ -0,0 +1,282 @@
+                    row.getVisibleCells(),
+                    (cell) => cell.id,
+                    (cell) => {
+                      console.log(cell);
```
Remove

matthewgallo reviewed (changes requested):
Just noticed some left over console logs. Something else is that the settings icon in the example with row selection has an incorrect fill color and also the popover opens in the wrong position, causes the page to jump to the incorrect position of the popover.

I also am seeing some style issues with `cds-table-batch-actions` but I also see that in the Carbon web component storybook.

devadula-nandan commented:
> Just noticed some left over console logs. Something else is that the settings icon in the example with row selection has an incorrect fill color and also the popover opens in the wrong position, causes the page to jump to the incorrect position of the popover.
> 
> I also am seeing some style issues with `cds-table-batch-actions` but I also see that in the Carbon web component storybook.

cleaned console logs (on batch actions as well)

fixed fill color (on batch actions as well)

i don't exactly know the root cause of why the popover opens in incorrect position, but i think its related to floating-menu used internally by overflow-menu

if there are more than 1 cds-table in the dom, we can notice this issue.

a recording of the issue seen in cds-table if there are more than 1 in the dom


https://github.com/user-attachments/assets/5356f2be-eb7f-4d12-86dd-c4f029d8642b

devadula-nandan commented on web-components/batch-actions/src/batch.ts:
```diff
@@ -261,13 +253,6 @@ export class MyBatchTable extends LitElement {
       place-items: center;
       flex-direction: column;
     }
-    cds-table-batch-actions[active] {
-      z-index: 1;
-      clip-path: polygon(0 0, 300% 0, 300% 300%, 0 300%);
-      opacity: 1;
-      pointer-events: all;
-      transform: translate3d(0, 0, 0);
-    }
```
i removed these styles as well on both examples, as i see they are already being applied. maybe carbon added these fixes??

matthewgallo commented:
Thanks for the research into the popover positioning. We should create an issue for them since it looks like the root cause is from within `@carbon/web-components` and not our examples here.
